### PR TITLE
Implements atomic infura API

### PR DESCRIPTION
**Pull request by clearloop**
## Desc

Use `AtomicUsize` to share the EthereumRPC thread-safely.

### CODE REVIEW

File: .github/workflows/rust.yml [hunk 1]
```diff
@@ -20,8 +20,8 @@ jobs:
...
           rustup component add clippy
       - name: Build
-        run: cargo build --all --all-features --all-targets --verbose
+        run: cargo build --all --verbose
       - name: Lint
-        run: cargo clippy --all --all-features --all-targets -- -D warnings
+        run: cargo clippy --all -- -D warnings
       - name: Run tests
-        run: cargo test --all --all-features --all-targets --verbose
+        run: cargo test --all --verbose
```
Comment (clearloop) [inline .github/workflows/rust.yml:27]: The `--all` flag contains `--all-features` and `--all-targets`, even docs

File: primitives/src/rpc/ethereum/mod.rs [hunk 2]
```diff
@@ -5,45 +5,79 @@ mod header;
...
+    /// let rpc = EthereumRPC::new(
+    ///   Client::new(),
+    ///   vec![
+    ///     "https://mainnet.infura.io/v3/0bfb9acbb13c426097aabb1d81a9d016".to_string(),
+    ///     "https://mainnet.infura.io/v3/74a9b1b5816b47aa853c23fcc4f2f3b6".to_string(),
+    ///   ],
+    /// );
+    ///
+    /// assert_ne!(rpc.rpc(), rpc.rpc());
+    /// ```
```
Comment (clearloop) [inline primitives/src/rpc/ethereum/mod.rs:47]: Tests here

File: primitives/src/rpc/ethereum/mod.rs [hunk 3]
```diff
@@ -5,45 +5,79 @@ mod header;
...
+    pub fn rpc(&self) -> &str {
+        self.atom.fetch_add(1, Ordering::SeqCst);
```
Comment (HackFisher) [inline primitives/src/rpc/ethereum/mod.rs:49]: Better to use self.atom.fetch_add(1, Ordering::SeqCst) % self.rpc.len() to avoid overflow?

**Review by HackFisher — APPROVED** [review]

File: primitives/src/rpc/ethereum/mod.rs (hunk 3 above)
Comment (clearloop) [reply primitives/src/rpc/ethereum/mod.rs:49]: `self.atom.fetch_add(1, Ordering::SeqCst) % self.rpc.len() ` is in the next line

File: primitives/rust-toolchain [hunk 4]
```diff
@@ -0,0 +1 @@
+nightly-2020-10-05
```
Comment (HackFisher) [inline primitives/rust-toolchain:1]: It seems latest nightly also works?

**Review by HackFisher — APPROVED** [review]

**Pull request merged**